python/ray/data/_internal/lazy_block_list.py

```diff
@@ -362,6 +362,37 @@ def __next__(self):
 
         return Iter()
 
+    def randomize_block_order(self, seed: Optional[int] = None):
+        """Randomizes the order of the blocks.
+
+        Args:
+            seed: Fix the random seed to use, otherwise one will be chosen
+                based on system randomness.
+        """
+        import random
+
+        if seed is not None:
+            random.seed(seed)
+
+        zipped = list(
+            zip(
+                self._tasks,
+                self._block_partition_refs,
+                self._block_partition_meta_refs,
+                self._cached_metadata,
+            )
+        )
+
+        random.shuffle(zipped)
+        tasks, block_partition_refs, block_partition_meta_refs, cached_metadata = map(
+            list, zip(*zipped)
+        )
+
+        self._tasks = tasks
```

Comment:
Instead of mutating the block list, how about returning a new one? Same for the non-lazy version.

Reply:
Woops yeah good call out.